darkcluster/src/test/java/com/linkedin/darkcluster/TestDarkClusterFilter.java

```diff
@@ -0,0 +1,107 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.darkcluster;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+import com.linkedin.common.util.Notifier;
+import com.linkedin.d2.balancer.Facilities;
+import com.linkedin.d2.balancer.LoadBalancerTestState;
+import com.linkedin.d2.balancer.simple.SimpleLoadBalancer;
+import com.linkedin.d2.balancer.util.ClusterInfoProvider;
+import com.linkedin.darkcluster.api.DarkClusterDispatcher;
+import com.linkedin.darkcluster.api.DarkClusterManager;
+import com.linkedin.darkcluster.api.DarkClusterStrategyFactory;
+import com.linkedin.darkcluster.api.DarkClusterVerifier;
+import com.linkedin.darkcluster.api.DarkClusterVerifierManager;
+import com.linkedin.darkcluster.api.NoOpDarkClusterVerifier;
+import com.linkedin.darkcluster.filter.DarkClusterFilter;
+import com.linkedin.darkcluster.impl.DarkClusterManagerImpl;
+import com.linkedin.darkcluster.impl.DarkClusterStrategyFactoryImpl;
+import com.linkedin.darkcluster.impl.DarkClusterVerifierManagerImpl;
+import com.linkedin.darkcluster.impl.DefaultDarkClusterDispatcherImpl;
+import com.linkedin.r2.filter.NextFilter;
+import com.linkedin.r2.message.RequestContext;
+import com.linkedin.r2.message.rest.RestRequest;
+import com.linkedin.r2.message.rest.RestRequestBuilder;
+import com.linkedin.r2.message.rest.RestResponse;
+import com.linkedin.r2.message.rest.RestResponseBuilder;
+
+import org.testng.annotations.Test;
+
+public class TestDarkClusterFilter
```

Comment:
Sorry I was unclear. Since this is a unit test we shouldn't be testing that the request gets sent. That's layers down. Tests should be just verifying that request was passed to both the managers when we expect. I notice you don't use Mocks much, that would make testing this a lot easier, as well as remove a lot of the setup code needed.